src/JsonApi/Serializer/ErrorNormalizer.php

```diff
@@ -49,6 +50,15 @@ public function normalize($object, $format = null, array $context = [])
             'description' => $this->getErrorMessage($object, $context, $this->debug),
         ];
 
+        $exceptionClass = get_class($object);
+        if ($object instanceof FlattenException || $object instanceof LegacyFlattenException) {
+            $exceptionClass = $object->getClass();
+        }
+
+        if (is_a($exceptionClass, ErrorCodeSerializableInterface::class, true)) {
+            $data['code'] = $exceptionClass::getErrorCode();
```

Comment:
I understand you're only interested in JSON:API, but this got me wondering, we should do something similar for other error formats, no?

Reply:
WDYT about extracting this as `getErrorCode` method in `ErrorNormalizerTrait`?

Reply:
sure if there are similar fields other formats.
in hydra I only find the "@id" field that can be used to reference an error but I think that's not what this is for. (http://www.hydra-cg.com/spec/latest/core/#description-of-http-status-codes-and-errors)
Maybe GraphQL has got something but I never used it before.

Reply:
For Hydra, `@id` seems like the way to go. But of course it should be a valid IRI.

Reply:
I could try to implement that but I have no idea how to generate a valid IRI for an error code,
that would imply that there are error resources with details that can be added etc. right?

Reply:
Hmm, yeah, I think something like that needs more thought. For now, let's just extract this method into the trait. :smile:

Reply:
something like:
```
ErrorNormalizerTrait
{
  function getErrorCode(\Exception $exception) : string
}
?

Reply:
There is already an `ErrorNormalizerTrait`: https://github.com/api-platform/core/blob/master/src/Problem/Serializer/ErrorNormalizerTrait.php

Sure, that method signature looks okay to me:

```php
private function getErrorCode(\Exception $exception): string
```

Reply:
I have updated the code now to use the trait.
I stuck with $object instead of \Exception $exception because FlattenExceptions dont seem to extend Exceptions.
I also made the return value nullable.